frontend/src/store/reducers/SignupReducer/ActionCreator.ts

```diff
@@ -1,35 +1,35 @@
 import { createAsyncThunk, SerializedError } from "@reduxjs/toolkit";
-import { ErrorResponse } from "@rtk-query/graphql-request-base-query/dist/GraphqlBaseQueryTypes";
-import { Exact, SignupMutation, UsersPermissionsRegisterInput } from "generated/graphql";
+import api from "api/axios";
+import { BASE_URL } from "config";
+import { IUser } from "types/User";
 
-interface SignupUserCreatorData {
-  signupMutationTrigger: (arg: Exact<{data: UsersPermissionsRegisterInput;}>) =>
-    Promise<{data: SignupMutation;} | {error: ErrorResponse | SerializedError;}>,
-  credentials: SignupUserCredentials, 
-}
 interface SignupUserCredentials {
   username: string,
   email: string,
   password: string,
+  role?: number,
+  confirmed?: boolean,
+  blocked?: boolean,
+}
+interface SignupUserResponse {
+  data: {
+    jwt: string,
+    user: IUser
+  }
 }
 
 export const signupUser = createAsyncThunk(
   'user/signup',
-  async (creatorData: SignupUserCreatorData, thunkAPI) => {
-    const {signupMutationTrigger, credentials} = creatorData;
-    const response = await signupMutationTrigger(
-      {data: {
-        username: credentials.username,
-        email: credentials.email,
-        password: credentials.password,
-      }}
-    );
+  async (creatorData: SignupUserCredentials, thunkAPI) => {
+    try {
+      const { data } = await api.post<SignupUserCredentials, SignupUserResponse>(`${BASE_URL}/api/auth/local/register`, creatorData);
```

Comment:
You may export axios like variables and use it here like axios.post it looks more familiar to a person working with a work on react))